File: python/armstrong_numbers.py
# How can you make this more scalable and reusable later?
def find_armstrong_numbers(numbers):

    armstrongs = []
    num_list = []

    if type(numbers) == int:
        num_list.append(numbers)
        numbers = num_list

      
    for num in numbers:

        if num < 10:
            armstrongs.append(num)

        else:
            digits = [int(x) for x in str(num)]
            power = len(digits)
            sum = 0

            for i in range(power):
                sum += digits[i]**power
            if sum == num:
                armstrongs.append(num)
    
    return armstrongs

File: js/armstrongNumbers.js
// How can you make this more scalable and reusable later?

exports.findArmstrongNumbers = function() {

};

function findArmstrongNumbers(numbers) {
    const armstrongs = [];

    for (const num of numbers) {
        if (num < 10) {
            armstrongs.push(num);
        } 
        
        else {
            const digits = num.toString().split('').map(Number);
            const power = digits.length;
            let sum = 0;

            for (let i = 0; i < power; i++) {
                sum += Math.pow(digits[i], power);
            }

            if (sum === num) {
                armstrongs.push(num);
            }
        }
    }

    return armstrongs;
}
